src/utils/builder.ts

```diff
@@ -329,25 +275,70 @@ export function setTransitionCondition(
   startNodeKey: string,
   transitionId: string,
   description: string,
-  elm: ElmLibrary
-): void {
+  elm: ElmLibrary,
+  criteria?: string
+): Pathway {
   const foundTransition = pathway.states[startNodeKey]?.transitions?.find(
     (transition: Transition) => transition.id === transitionId
   );
 
+  const cql = criteria ? criteria : getElmStatement(elm).name;
+
   if (foundTransition)
     foundTransition.condition = {
       description: description,
-      cql: getElmStatement(elm).name,
+      cql: cql,
       elm: elm
     };
+
+  return {
+    ...pathway,
+    states: {
+      ...pathway.states,
+      [startNodeKey]: {
+        ...pathway.states[startNodeKey]
+      }
+    }
+  };
 }
 
 export function setGuidanceStateElm(pathway: Pathway, key: string, elm: ElmLibrary): void {
   (pathway.states[key] as GuidanceState).elm = elm;
   (pathway.states[key] as GuidanceState).cql = getElmStatement(elm).name;
 }
 
+export function setTransitionCql(
+  pathway: Pathway,
+  criteria: string,
+  transitionId: string,
+  startNodeKey: string
```

Comment:
This is super minor but all of the other methods here follow the format `(pathway, keys/ids, new value)`. Also while we may be passing in the value `criteria.label` when we call the method I think it makes more sense to name the argument `cql` instead of `criteria`

Reply:
I ended up not needing this once I incorporated 261, going to remove it.